SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp

```diff
@@ -81,267 +81,346 @@ SpatialFunctionalTestRequireHandler::SpatialFunctionalTestRequireHandler()
+
+	return GenericRequire(Msg, bCheckTrue, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireFalse(bool bCheckFalse, const FString& Msg)
 {
-	return GenericRequire(Msg, !bCheckFalse, FString());
+	FString StatusMsg = FString::Printf(TEXT("Received False as expected"));
+
+	return GenericRequire(Msg, !bCheckFalse, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireCompare(int A, EComparisonMethod Operator, int B, const FString& Msg)
 {
 	bool bPassed = Compare(A, Operator, B);
-
-	FString ErrorMsg;
+	FString StatusMsg;
+	FString OperatorStr = GetComparisonMethodAsString(Operator);
 
 	if (!bPassed)
 	{
-		FString OperatorStr = GetComparisonMethodAsString(Operator);
-		ErrorMsg = FString::Printf(TEXT("Received %d %s %d but was expecting A %s B"), A, *OperatorStr, B, *OperatorStr);
+		StatusMsg = FString::Printf(TEXT("Received %d %s %d but was expecting A %s B"), A, *OperatorStr, B, *OperatorStr);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %d %s %d as expected"), A, *OperatorStr, B);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireCompare(float A, EComparisonMethod Operator, float B, const FString& Msg)
 {
 	bool bPassed = Compare(A, Operator, B);
-
-	FString ErrorMsg;
+	FString StatusMsg;
+	FString OperatorStr = GetComparisonMethodAsString(Operator);
 
 	if (!bPassed)
 	{
-		FString OperatorStr = GetComparisonMethodAsString(Operator);
-		ErrorMsg = FString::Printf(TEXT("Received %f %s %f but was expected A %s B"), A, *OperatorStr, B, *OperatorStr);
+		StatusMsg = FString::Printf(TEXT("Received %f %s %f but was expected A %s B"), A, *OperatorStr, B, *OperatorStr);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %f %s %f as expected"), A, *OperatorStr, B);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireEqual(bool bValue, bool bExpected, const FString& Msg)
 {
 	bool bPassed = bValue == bExpected;
-	FString ErrorMsg;
+	FString StatusMsg;
+	FString ValueStr = bValue ? TEXT("True") : TEXT("False");
 
 	if (!bPassed)
 	{
-		FString ValueStr = bValue ? TEXT("True") : TEXT("False");
 		FString ExpectedStr = bExpected ? TEXT("True") : TEXT("False");
-		ErrorMsg = FString::Printf(TEXT("Received %s but was expecting %s"), *ValueStr, *ExpectedStr);
+		StatusMsg = FString::Printf(TEXT("Received %s but was expecting %s"), *ValueStr, *ExpectedStr);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %s as expected"), *ValueStr);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireEqual(int Value, int Expected, const FString& Msg)
 {
 	bool bPassed = Value == Expected;
-	FString ErrorMsg;
+	FString StatusMsg;
```

Comment:
Very minor but for extra cpp cool points you could write:
```
const FString StatusMsg = [](&Value, &Expected){
	if (bPassed)
	{
		return FString::Printf(TEXT("Received %d as expected"), Value);
	}
	return FString::Printf(TEXT("Received %d but was expecting %d"), Value, Expected);
}();
```

Reply:
Or maybe write this lambda as a free function as it you do it a few times.

Reply:
The lambda is slightly different for each of the requires so it wouldn't be useful as a free function. Is there any performance/readability benefit to writing it as a lambda?

Reply:
StatusMsg gets to be const which is nice. Its all wrapped into a single assignment which I also like. It uses return statements for control flow which I think is more explicit than if branches.

Reply:
I also think you could still totally have a free function. Mostly you have this pattern: 
`StatusMsg = FString::Printf(TEXT("Received %s but was expecting %s"), *Value, *Expected);`
And while some of the transformations are different, if you write a free function that shouldn't be an issue. With a bit of work and a optional parameter you could cover all your cases.
```
FString makeStatusString(bool passed, FString received, FString expected, tolerance = FString())
{
	if(passed)
	{
		if (tolerance.isEmpty())
		{
			return FString::Printf(TEXT("Received %d as expected"), received);
		}
		return FString::Printf(TEXT("Received %f as expected (within tolerance %f of %f)"), received, tolerance, expected);
	}
	else
	{
		if (tolerance.isEmpty())
		{
			return FString::Printf(TEXT("Received %s but was expecting %s"), received, expected);
		}
		return FString::Printf(TEXT("Received %f but was expecting %f (tolerance %f)"), expected, expected, tolerance);
	}
}
```